gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateConfig.groovy

```diff
@@ -208,8 +211,12 @@ class GateConfig extends RedisHttpSessionConfiguration {
   }
 
   @Bean
-  @ConditionalOnProperty('services.kayenta.enabled')
-  KayentaService kayentaService(OkHttpClient okHttpClient) {
+  @ConditionalOnProperty([ 'services.kayenta.enabled', 'services.kayenta.externalhttps' ])
+  KayentaService kayentaService(OkHttpClientConfigurationProperties props, OkHttpMetricsInterceptor interceptor) {
```

Comment:
Unless there is logic elsewhere that produces a `KayentaService` bean, won't gate fail to start when `services.kayenta.enabled` is `true` (and `services.kayenta.externalhttps` is not)?

Reply:
it seems like what you want here is a parameter to the method `@Value('services.kayenta.externalhttps') boolean externalHttps` and use that to conditionally customize the okHttpClient reconfiguration

Reply:
OK, adjusted.  Can y'all take another quick peek?

Reply:
LGTM. I didn't pull it down and run it though.

Reply:
Would like @cfieber to eyeball as well.